### PR TITLE
Decorate a class

### PR DESCRIPTION
**Pull request by JavierAybar**
# **_`OOP school library: decorate a class`_** 📚 
_**In this project I did :**_

✅ _**Create a class `Nameable`**_.
✅ **_Turn your `Person` class to `Nameable`_**.
✅ **_Prepare base `Decorator`_**.
✅ **_Prepare `CapitalizeDecorator` and `TrimmerDecorator`_**.

## **_You can see the project requirements [here](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_decorator_v2.md)_**

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -13,9 +50,20 @@ def can_use_services?
...
     @age >= 18
   end
 end
+
+# person = Person.new(22, 'maximilianus')
+# person.correct_name
+# capitalized_person = CapitalizeDecorator.new(person)
+# puts capitalized_person.correct_name
+# capitalized_trimmed_person = TrimmerDecorator.new(capitalized_person)
+# puts capitalized_trimmed_person.correct_name
```
Comment (MahmoudBakr23) [inline person.rb]: - [ ] Please delete any unused comments for a cleaner code base and more optimized app. 🙏

File: person.rb [hunk 2]
```diff
@@ -1,9 +1,46 @@
...
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
```
Comment (MahmoudBakr23) [inline person.rb]: - [ ] In the Decorator class, please specify the relationship between it and the Nameable class by defining the `attr_accessor` before the `initialize` method 🙏

File: person.rb [hunk 3]
```diff
@@ -1,9 +1,46 @@
-class Person
```
Comment (MahmoudBakr23) [inline person.rb:1]: - [ ] Please specify the `Person` class with the `correct_name` method in it as per project requirements. 🙏

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hello @JavierAybar 😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Well done on creating the classes and custom methods so far
- Descriptive PR and README
- Correct Linters

However, there is always room for improvement.

> I'd like to remind you that your long hours of work are appreciated. However, the code review process is for improving your code, not criticizing it. 😄

 ## Required Changes ♻️
_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

- [ ] Please keep in mind that it'd be way better if you divided your classes into multiple files for each class. 🙏 

Cheers, and Happy coding!👏👨‍💻💻

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 3 above)
Comment (JavierAybar) [reply person.rb:1]: @MahmoudBakr23 
Sorry but I don't see the error in my code, I think it fulfilled that 
https://github.com/JavierAybar/OOP-school-library/blob/b1dd42a54372b4558f8d563ff029b48e58d23ba7/person.rb#L53-L55

Requirement
Make sure that this class has a method correct_name implemented. It should simply return the name attribute.

File: person.rb (hunk 2 above)
Comment (JavierAybar) [reply person.rb]: I don't see the need to use an attr_accessor here and I don't see it specified in the requirements either.
Or is your suggestion to make it more readable?

**Review by joskalenda — APPROVED** [review]
| STATUS: APPROVED ✅ |
|------------------------|
| Hi @JavierAybar , 🥰|
| Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: Congratulations! 🎉  |
|  |

### Highlights
✅ ... Green linters
✅ ... Good commit messages

---------
### Optional suggestions

> _Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

### Cheers and Happy coding!👏👏👏
-------
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me(@joskalenda) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**